cico_functions.sh

```diff
@@ -108,13 +108,27 @@ function build_and_push() {
     tag_push "${REGISTRY}/${ORGANIZATION}/${IMAGE}:${TAG}"
     echo "CICO: '${TAG}'  version of images pushed to '${REGISTRY}/${ORGANIZATION}' organization"
   fi
+}
 
-  # Let's now build and push arbitrary-user patched images only to 'eclipse' quay.io organization
-  # which is done as part of the 'centos' target execution
-  if [ "$TARGET" == "centos" ]; then
-    # If additional tag (e.g. "nightly") is NOT set let's use git commit hash as a tag
-    export TAG=${TAG:-${GIT_COMMIT_TAG}}
-    "${SCRIPT_DIR}"/arbitrary-users-patch/build_images.sh --push
-    echo "CICO: pushed '${TAG}' version of the arbitrary-user patched base images"
-  fi
+# Build release version of devfile registry, using ${TAG} as a tag. For release
+# versions, the devfiles are rewritten to refer to ${TAG}-tagged images with the
+# arbitrary user patch
+function build_and_push_release() {
+  echo "CICO: building release '${TAG}' version of devfile registry"
+  docker build -t ${IMAGE} -f ${DOCKERFILE} . \
+    --build-arg PATCHED_IMAGES_REG=${REGISTRY} \
+    --build-arg PATCHED_IMAGES_ORG=${ORGANIZATION} \
+    --build-arg PATCHED_IMAGES_TAG=${TAG}
+  echo "CICO: release '${TAG}' version of devfile registry built"
+  tag_push "${REGISTRY}/${ORGANIZATION}/${IMAGE}:${TAG}"
+  echo "CICO: release '${TAG}' version of devfile registry pushed to '${REGISTRY}/${ORGANIZATION}' organization"
+}
+
+# Build images patched to work on OpenShift (work with arbitrary user IDs) and push
+# them to registry
+function build_patched_base_images() {
```

Comment:
Just double-checking - This will be executed only for `centos` target right? Since push with `rhel` is going to fail

Reply:
or most like during the `release` CI images would be build & pushed twice to the public quay.io `eclipse` registry

Reply:
It would execute during rhel builds as well -- I suppose more attention should be given to this case; it's confusing since we're trying to support four flows through the same code based off environment variables.

When the release script is called with `TARGET='rhel'`, the changes in this PR would 
1. Build arbitrary-user-id patched base images for `quay.io/openshiftio/<image>:<release-tag>` (these repos don't currently exist and so the build **would fail**)
2. Build a devfile registry `quay.io/openshiftio/rhel-che-devfile-registry:<release>` that references `quay.io/openshiftio/<image>:<release>`

For nightly `rhel`-target builds, the build should be unmodified from previous.

Reply:
I could rework the CI build to use the `eclipse` org for all patched images (so that rhel builds reference `quay.io/eclipse/<image>:<release>`) but that would involve more complicated logic. 

Currently I've tried to structure the CI so that once the environment variables are set, the same `REPOSITORY`, `ORGANIZATION`, and `TAG` are used for both patched images and the built registry.